mrtt-ui/src/validation/multiSelectWithOther.js

```diff
@@ -0,0 +1,28 @@
+import language from '../language'
+import * as yup from 'yup'
+
+export const otherIsCheckedButInputIsEmptyValidation = (schema) =>
+  schema.test({
+    name: 'isOtherCheckedAndEmpty',
+    message: language.multiselectWithOtherFormQuestion.validation.clairfyOther,
+    test: (value, context) => {
+      const {
+        parent: { otherValue }
+      } = context
+      return otherValue !== undefined && otherValue !== ''
+    }
+  })
+
+export const multiselectWithOtherValidation = yup.object({
+  selectedValues: yup
+    .array()
+    .of(yup.string())
+    .when('isOtherChecked', {
+      is: false || undefined,
+      then: (schema) =>
+        schema.min(1, language.multiselectWithOtherFormQuestion.validation.selectAtleastOneAim),
```

Comment:
Awesome possum. The language coming from the language file is specific to aims though. Can we make it generic via a couple of parameters (or more ideally an options parameter)?